fakes_test.go

```diff
@@ -24,16 +62,56 @@ func Test_VirtualMachinesClient_Get(t *testing.T) {
 
 	// the fake VM must return the provided name and its ID contain the provided resource group name.
 
-	// TODO: populate vm with response from fake
+	responseVM := armcompute.VirtualMachine{
+		Name: to.Ptr[string](vmName),
+		ID:   to.Ptr[string](resourceGroupName), // This isn't a real ID but good enough for our purposes
+	}
+
 	var vm armcompute.VirtualMachine
 
+	// matthchr: It wasn't clear to me until I read the README in armcompute that I needed to embed the fake into my own type,
+	// I had guessed at that but then though I was missing something because it's going to make things pretty verbose (need to define a new type for each test).
```

Comment:
I had envisioned that tests would reuse a fake across multiple tests.  You bring up a good point though that it does mean the fake response is essentially hard-coded.  i.e. I can't have two tests that call `Get()` each with their own fake response.  Let me think about this some more as I agree it's a problem.

Reply:
> I can't have two tests that call Get() each with their own fake response.

I mean, you can if you create your fake in such a way that when you create it you pass the vm to be returned. But depending on how much you wanna mock you end up doing some legwork to do that for each API you're mocking.